recipes/boost/all/conanfile.py

```diff
@@ -385,6 +386,13 @@ def validate(self):
                     if min_compiler_version is not None:
                         if tools.Version(self.settings.compiler.version) < min_compiler_version:
                             raise ConanInvalidConfiguration("Boost.Math requires a C++11 capable compiler")
+                            
+        if not self.options.without_python:
+            #Boost python library build will fail if python package numpy is not installed
+            try:
+                import numpy
```

Comment:
what if we want to use different Python that was given through "python_executable" for example?

Reply:
You should use the run python script helper to test for this

Reply:
the problem is with `CONAN_PRINT_RUN_COMMANDS`(well-known issue: https://github.com/conan-io/conan/issues/8319 https://github.com/conan-io/conan/issues/6752)
seems like it's not going to be fixed, so I guess we have to add a workaround to `_run_python_script`

Reply:
actually, there is already a workaround for that: https://github.com/conan-io/conan-center-index/pull/5643/files#diff-cc14160728d1ad321c1444ffad82615139181e681f4acb7dfd045a005cc8bd66R487
seems like it doesn't always work?

Reply:
Even when python numpy is installed, when I build the boost package with conan create using VS2019 I get unresolved symbol link errors on boost.numpy for the numpy.cpp executable in the test package.
So better to fix the numpy problems in a separate PR, I removed the numpy check that I added from this PR.